// Copyright 2022 The urpc Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package urpc.protocol.urpc;

message RPCMeta {
    RequestMeta request = 1;
    ResponseMeta response = 2;
    int64 correlation_id = 3;
    int32 attachment_size = 4;
}

message RequestMeta {
    string service_name = 1;
    string method_name = 2;
    int64 log_id = 3;
}

message ResponseMeta {
    int32 error_code = 1;
    optional string error_text = 2;
}
